smolyay/samples.py

```diff
@@ -0,0 +1,409 @@
+import abc
+import numpy
+
+
+class UnidimensionalPointSet(abc.ABC):
+    """Set of unidimensional points
+
+    A set of unique unidimensional points within a domain.
+    """
+
+    def __init__(self, domain):
+        self._points = None
+        self._domain = numpy.array(domain)
+        self._valid_cache = False
+
+    @property
+    def domain(self):
+        """numpy.ndarray: Domain of the `points`."""
+        return self._domain
+
+    @property
+    def points(self):
+        """numpy.ndarray: Points in the set."""
+        if not self._valid_cache:
+            self._create()
+        self._valid_cache = True
```

Comment:
Put this inside the if, it only needs to run if the cache was filled.
```suggestion
            self._valid_cache = True
```